### PR TITLE
feat: integrate World ID for event registration in EventManager contract

### LINKED ISSUE #4: Register for Event
- Integrate World ID on-chain verification for event registration.
- Build the registerEvent function to allow participants to register for an event.
- Write unit tests for the registerEvent function.

### PR DESCRIPTION
**Pull request by Jagadeeshftw**
### Summary  
This PR integrates **World ID** for event registration in the `EventManager` contract to enhance security and prevent duplicate event submissions using nullifier hashes. closes #4 

### Checklist  
- [x] World ID integrated successfully.  
- [x] Tests updated to validate new functionality.  
- [x] Obsolete code removed.

### Testing  
All tests have been updated and passed successfully.
image

### CODE REVIEW

**Comment by Jagadeeshftw** [comment]
@Ugo-X This PR is ready for review

**Comment by Ugo-X** [comment]
Please add event registration for participants, not for event creation.

As event creation is already another issue

**Comment by Jagadeeshftw** [comment]
Your comments were addressed @Ugo-X, tests are failing because the event is not yet created.

**Comment by Ugo-X** [comment]
> Your comments were addressed @Ugo-X, tests are failing because the event is not yet created.

give me some time, will merge

File: src/EventManager.sol [hunk 1]
```diff
@@ -1,28 +1,100 @@
...
+    /// @dev Register a participant for an existing event.
+    /// @param eventId The ID of the event to register for.
+    function registerParticipant(uint256 eventId) public {
+        require(eventId < nextEventId, "Event does not exist");
+        require(!registeredParticipants[eventId][msg.sender], "Already registered as participant");
+
+        events[eventId].participants.push(msg.sender);
+        registeredParticipants[eventId][msg.sender] = true;
+
+        emit ParticipantRegistered(eventId, msg.sender);
```
Comment (Ugo-X) [inline src/EventManager.sol:33]: Worldid verification is not implemented,  we only want users to be able to register after they have verified with worldid ...

**Comment by Ugo-X** [comment]
@Jagadeeshftw what is the update?

**Comment by Jagadeeshftw** [comment]
Working on it @Ugo-X

**Comment by Ugo-X** [comment]
> Working on it @Ugo-X

There are also conflicts to be resolved

**Comment by Ugo-X** [comment]
Screenshot 2024-12-15 at 13 42 20

**Review by Ugo-X — COMMENTED** [review]
still conflicts on this PR

**Pull request merged**